src/server/index_patterns/service/lib/index.js

```diff
@@ -1,3 +1,3 @@
 export { getFieldCapabilities } from './field_capabilities';
 export { resolveTimePattern } from './resolve_time_pattern';
-export { createNoMatchingIndicesError, isNoMatchingIndicesError } from './errors';
```

Comment:
Yep, it's not in `src` unfortunately [test/api_integration/apis/index_patterns/es_errors/errors.js](https://github.com/elastic/kibana/blob/master/test/api_integration/apis/index_patterns/es_errors/errors.js)